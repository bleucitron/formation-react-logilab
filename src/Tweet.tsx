import React from 'react';
import { FaTwitter } from 'react-icons/fa';

interface User {
  name: string;
}

export interface ITweet {
  id: number;
  text: string;
  created_at: string;
  lang: string;
  user: User;
}

type TweetProps = ITweet;

function Tweet(props: TweetProps) {
  const { text, created_at: date, user } = props;

  return (
    <li className='Tweet'>
      <div className='text'>{text}</div>
      <div className='date'>{date}</div>
      <div className='user'>{user.name}</div>
      <FaTwitter />
    </li>
  );
}

export default Tweet;
