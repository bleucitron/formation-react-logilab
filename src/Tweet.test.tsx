import React from 'react';
import { mount, shallow } from 'enzyme';
import Tweet from './Tweet';
import tweets from './data/tweets.json';

describe('Tweet', () => {
  it('should not crash on mount', () => {
    mount(<Tweet {...tweets[0]} />);
  });

  it('should match snapshots', () => {
    const wrapper = shallow(<Tweet {...tweets[0]} />);

    expect(wrapper).toMatchSnapshot();
  });
});
