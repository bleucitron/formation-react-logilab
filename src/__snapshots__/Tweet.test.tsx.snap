// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet should match snapshots 1`] = `
<li
  className="Tweet"
>
  <div
    className="text"
  >
    RT @benoithamon: Rien n'y fait. La condamnation de Monsanto par la justice américaine, la multiplication des cancers chez les enfants, les…
  </div>
  <div
    className="date"
  >
    Sat Sep 15 07:09:51 +0000 2018
  </div>
  <div
    className="user"
  >
    Romain l'Ourson
  </div>
</li>
`;
